backends/tfhe-cuda-backend/cuda/src/integer/cast.cuh

```diff
@@ -9,10 +9,29 @@ template <typename Torus>
 __host__ void host_extend_radix_with_trivial_zero_blocks_msb(
     CudaRadixCiphertextFFI *output, CudaRadixCiphertextFFI const *input,
     cudaStream_t const *streams, uint32_t const *gpu_indexes) {
-
   copy_radix_ciphertext_slice_async<Torus>(streams[0], gpu_indexes[0], output,
                                            0, input->num_radix_blocks, input, 0,
                                            input->num_radix_blocks);
 }
 
+template <typename Torus>
+__host__ void host_trim_radix_blocks_lsb(CudaRadixCiphertextFFI *output,
+                                         CudaRadixCiphertextFFI const *input,
+                                         cudaStream_t const *streams,
+                                         uint32_t const *gpu_indexes) {
+
+  const uint32_t input_start_lwe_index =
+      input->num_radix_blocks - output->num_radix_blocks;
+
+  if (input_start_lwe_index < 0 ||
```

Comment:
This is not the right check here: `input_start_lwe_index` is a uint variable so it can't be negative. Instead we should check: 
```
if (input->num_radix_blocks <= output->num_radix_blocks)
    PANIC("Cuda error: input num blocks should be greater than output num blocks");